File: backend/app/data/generator_types.py
"""Kanonische Erzeugercodes für Berechnung, Legende und Vektorexport."""

GENERATOR_TYPE_LABELS = {
    "ews_wp": "Sole/Wasser-WP (Erdsonden)",
    "lwwp": "Luft/Wasser-WP",
    "wasser_wp": "Wasser/Wasser-WP",
    "co2_wp": "CO₂-Wärmepumpe",
    "fernwaerme": "Fernwärme",
    "gas": "Gaskessel",
    "oel": "Ölkessel",
    "holz": "Holz-/Pelletkessel",
    "elektro": "Elektroheizung",
    "hybrid": "Hybridanlage",
    "sonstige": "Sonstiger Wärmeerzeuger",
}

HEAT_PUMP_TYPES = {"ews_wp", "lwwp", "wasser_wp", "co2_wp", "hybrid"}
SOURCE_CIRCUIT_TYPES = {"ews_wp", "wasser_wp"}


def generator_type_label(value: str | None) -> str | None:
    if not value:
        return None
    return GENERATOR_TYPE_LABELS.get(str(value), str(value))

File: frontend/src/components/hc/nodes/generatorTypes.js
// Eine zentrale Quelle für Erzeugercodes, UI-Bezeichnungen und Fachgruppen.
// Die Codes sind identisch mit ProjectContext, LV-Import und Kostenschätzung.
export const GENERATOR_TYPES = [
  { value:'ews_wp',     label:'Sole/Wasser-WP (Erdsonden)', family:'wp', source:'sole' },
  { value:'lwwp',       label:'Luft/Wasser-WP', family:'wp', source:'air' },
  { value:'wasser_wp',  label:'Wasser/Wasser-WP', family:'wp', source:'water' },
  { value:'co2_wp',     label:'CO₂-Wärmepumpe', family:'wp', source:'co2' },
  { value:'fernwaerme', label:'Fernwärme', family:'district' },
  { value:'gas',        label:'Gaskessel', family:'boiler', fuel:'Gas' },
  { value:'oel',        label:'Ölkessel', family:'boiler', fuel:'Öl' },
  { value:'holz',       label:'Holz-/Pelletkessel', family:'boiler', fuel:'Holz' },
  { value:'elektro',    label:'Elektroheizung', family:'electric' },
  { value:'hybrid',     label:'Hybridanlage', family:'hybrid' },
  { value:'sonstige',   label:'Sonstiger Wärmeerzeuger', family:'other' },
];

export const GENERATOR_TYPE_LABELS = Object.fromEntries(
  GENERATOR_TYPES.map(item => [item.value, item.label])
);

export const LWWP_BAUARTEN = [
  { value:'aussenaufstellung', label:'Monoblock – Aussenaufstellung' },
  { value:'innenaufstellung', label:'Monoblock – Innenaufstellung' },
  { value:'split', label:'Splitgerät – Aussen- und Inneneinheit' },
];

export const generatorType = value =>
  GENERATOR_TYPES.find(item => item.value === value) || null;

export const istWaermepumpe = value =>
  ['wp', 'hybrid'].includes(generatorType(value)?.family);
export const hatSoleOderWasserkreis = value => ['ews_wp', 'wasser_wp'].includes(value);
